pkg/runtime/framework/core/framework_test.go

```diff
@@ -498,9 +500,14 @@ func TestRunComponentBuilderPlugins(t *testing.T) {
 		t.Run(name, func(t *testing.T) {
 			ctx, cancel := context.WithCancel(context.Background())
 			t.Cleanup(cancel)
+
 			clientBuilder := testingutil.NewClientBuilder()
+			if tc.trainingRuntime != nil {
+				clientBuilder.WithObjects(tc.trainingRuntime)
```

Comment:
Uhm, is this bug, previously?

Reply:
Before the `runtimeJobTemplate` was provided "pre-computed", now it's really taken from `trainingRuntime` based on the new `RunComponentBuilderPlugins` signature.

Reply:
I see. Thank you for clarifying. I confirmed the JobSet plugin. TBH, I'm currently thinking re-add object to Task struct like https://github.com/kubeflow/trainer/blob/62a058db0c445db1b7e16572d83fdace56f59476/pkg/runtime.v2/runtime.go#L36

This could reduce API each plugins. But I do not offer to re-add it in this PR.
If you decide not to work on re-adding it in this PR, I will work on it in a separate PR.

Reply:
Ah actually `pkg/runtime/runtime.go` is very interesting because it was somehow already geared towards apply configuration, with all these `With*` functions like `WithLabels`, `WithPodSpecReplicas`, ...

I think it needs a bit more thoughts. So I'd also suggest we can do it in a follow up PR.

Reply:
yeah, I agree with you. We need to rework pkg/runtime/runtime.go.
I will try to see if we can improve that.